ffmuc-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -28,22 +28,31 @@ clean_port() {
 	echo "$1" | sed -r 's/:[0-9]+$|\[|\]//g'
 }
 
-check_address_family() {
+extract_port() {
+    local address_port="$1"
+
+    # Remove brackets for IPv6 addresses
+    local address="$(echo "$address_port" | sed -e 's/\[//;s/\]//')"
+
+    # Extract port number
+    echo "$address" | awk -F: '{print $NF}'
+}
+
+resolve_host() {
 	local peer_endpoint="$2"
 	local gateway
 	gateway="$(clean_port "$peer_endpoint")"
 	# Check if we have a default route for v6 if not fallback to v4
 	if ip -6 route show table 1 | grep -q 'default via'
 	then
 		local ipv6
-		ipv6="$(gluon-wan nslookup "$gateway" | grep 'Address:\? [0-9]' | grep -E -o '([a-f0-9:]+:+)+[a-f0-9]+')"
-		echo "[$ipv6]$(echo "$peer_endpoint" | grep -E -oe ":[0-9]+$")"
+		ipv6="$(gluon-wan nslookup "$gateway" | grep 'Address:\? [0-9]' | grep -oE '([a-f0-9:]+:+)+[a-f0-9]+')"
+		echo "$ipv6"
```

Comment:
Assuming this regex magic still works, drop the variable `ipv6`.
```suggestion
		gluon-wan nslookup "$gateway" | grep 'Address:\? [0-9]' | grep -oE '([a-f0-9:]+:+)+[a-f0-9]+'
```

Reply:
I find the explicit `echo` quite nice for signalling where data is returned, instead of having to guess where the return value (or whether this function is even supposed to return something).
But I'm okay with removing it if you prefer.